src/Util/File.cc

```diff
@@ -67,3 +67,89 @@ fs::path SearchPath(std::string filename) {
     }
     return fs::path();
 }
+
+CARTA::FileType GuessImageType(const std::string& path_string, bool check_content) {
+    if (check_content) {
+        // Guess file type by magic number
+        auto magic_number = GetMagicNumber(path_string);
+        switch (magic_number) {
+            case FITS_MAGIC_NUMBER:
+                return CARTA::FITS;
+            case HDF5_MAGIC_NUMBER:
+                return CARTA::HDF5;
+            case GZ_MAGIC_NUMBER:
+                fs::path gz_path(path_string);
```

Comment:
Duplicates code in IsCompressedFits (in this file), which also checks the magic number but only checks for lower case ".fits".

Reply:
ok, I've made the two functions consistent now.